### PR TITLE
Disable ConnectWithRevocation_WithCallback test

### PR DESCRIPTION
**Pull request by rzikm**
Contributes to #74034.

### CODE REVIEW

**Review by CarnaViire — APPROVED** [review]
LGTM, thanks!

**Review by CarnaViire — CHANGES_REQUESTED** [review]
Note Jeremy's comment on the issue:

> I only see the failures occurring on OSX, so it should only be disabled on OSX, not all Unixes. (The Linuxes are passing stably)

File: src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs [hunk 1]
```diff
@@ -96,6 +96,7 @@ public async Task DefaultConnect_EndToEnd_Ok(string host)
         [InlineData(true)]
         [InlineData(false)]
         [SkipOnPlatform(TestPlatforms.Android, "The invalid certificate is rejected by Android and the .NET validation code isn't reached")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/74034", TestPlatforms.AnyUnix)]
```
Comment (danmoseley) [inline src/libraries/System.Net.Security/tests/FunctionalTests/CertificateValidationRemoteServer.cs:99]:
```suggestion
        [ActiveIssue("https://github.com/dotnet/runtime/issues/74034", TestPlatforms.OSX)]
```

**Comment by danmoseley** [comment]
I edited to OSX so validation could rerun. note that doesn't include iOS and tvOS, not sure whether those are relevant.

**Comment by rzikm** [comment]
> I edited to OSX so validation could rerun. note that doesn't include iOS and tvOS, not sure whether those are relevant.

I am not aware of any failures on those, so it either works there or is disabled by some other means.

**Review by CarnaViire — APPROVED** [review]

**Comment by rzikm** [comment]
CI failures are weird, but build succeeded and the test itself is actually not outerloop so any failures are not result of this change.

**Pull request merged**